src/multi.c

```diff
@@ -65,9 +66,17 @@ void queueMultiCommand(client *c, uint64_t cmd_flags) {
      * aborted. */
     if (c->flags & (CLIENT_DIRTY_CAS|CLIENT_DIRTY_EXEC))
         return;
-
-    c->mstate.commands = zrealloc(c->mstate.commands,
-            sizeof(multiCmd)*(c->mstate.count+1));
+    if (c->mstate.count == 0) {
+        /* If a client is using multi/exec, assuming it is used to execute at least
+         * two commands. Hence, creating by default size of 2. */
+        c->mstate.commands = zmalloc(2 * sizeof(multiCmd));
+        c->mstate.alloc_count = 2;
+    }
+    if (c->mstate.count == c->mstate.alloc_count) {
+        c->mstate.commands = zrealloc(c->mstate.commands,
+                sizeof(multiCmd)*(c->mstate.alloc_count*2));
```

Comment:
maybe after a certain threshold we should start adding a fixed extra rather than grow to *2? same as we do in sdsMakeRoomFor?

on the other hand, i see that we did not choose to do that in the incremental growth of argv (see argv_len)
but i see we did add some overflow protection there.

Reply:
I did think about linear growth but wasn't sure about the threshold hence opted for this simple approach. Thanks for the suggestion around overflow protection adding that.